src/FollowMe/FollowMe.cc

```diff
@@ -0,0 +1,218 @@
+        _gcsMotionReportTimer.stop();
+    }
+}
+
+void FollowMe::_setGPSLocation(QGeoPositionInfo geoPositionInfo)
+{
+    if (geoPositionInfo.isValid())
+    {
+        // get the current location coordinates
+
+        QGeoCoordinate geoCoordinate = geoPositionInfo.coordinate();
+
+        _motionReport.lat_int = geoCoordinate.latitude()*1e7;
+        _motionReport.lon_int = geoCoordinate.longitude()*1e7;
+        _motionReport.alt = geoCoordinate.altitude();
+
+#ifdef QT_QML_DEBUG
+        if(_simulate_motion == true) {
+            _motionReport.lat_int = 47.3977420*1e7;
+            _motionReport.lon_int = 8.5455941*1e7;
+            _motionReport.alt = 488.00;
+        }
+#endif
+        // get the current eph and epv
+
+        if(geoPositionInfo.hasAttribute(QGeoPositionInfo::HorizontalAccuracy) == true) {
+            _motionReport.pos_std_dev[0] = _motionReport.pos_std_dev[1] = geoPositionInfo.attribute(QGeoPositionInfo::HorizontalAccuracy);
+        }
+
+        if(geoPositionInfo.hasAttribute(QGeoPositionInfo::VerticalAccuracy) == true) {
+            _motionReport.pos_std_dev[2] = geoPositionInfo.attribute(QGeoPositionInfo::VerticalAccuracy);
+        }
+
+        // calculate z velocity if it's availible
+
+        if(geoPositionInfo.hasAttribute(QGeoPositionInfo::VerticalSpeed)) {
+            _motionReport.vz = geoPositionInfo.attribute(QGeoPositionInfo::VerticalSpeed);
+        }
+
+        // calculate x,y velocity if it's availible
+
+        if((geoPositionInfo.hasAttribute(QGeoPositionInfo::Direction)   == true) &&
+           (geoPositionInfo.hasAttribute(QGeoPositionInfo::GroundSpeed) == true)) {
+
+            qreal direction = _degreesToRadian(geoPositionInfo.attribute(QGeoPositionInfo::Direction));
+            qreal velocity = geoPositionInfo.attribute(QGeoPositionInfo::GroundSpeed);
+
+            _motionReport.vx = cos(direction)*velocity;
+            _motionReport.vy = sin(direction)*velocity;
+        }
+    }
+}
+
+void FollowMe::_sendGCSMotionReport(void)
+{
+    QmlObjectListModel & vehicles = *_toolbox->multiVehicleManager()->vehicles();
+    MAVLinkProtocol* mavlinkProtocol = _toolbox->mavlinkProtocol();
+    mavlink_follow_target_t follow_target;
+
+    memset(&follow_target, 0, sizeof(mavlink_follow_target_t));
+
+    follow_target.timestamp = runTime.nsecsElapsed()*1e-6;
+    follow_target.est_capabilities = (1 << POS);
+    follow_target.position_cov[0] = _motionReport.pos_std_dev[0];
+    follow_target.position_cov[2] = _motionReport.pos_std_dev[2];
+    follow_target.alt = _motionReport.alt;
+    follow_target.lat = _motionReport.lat_int;
+    follow_target.lon = _motionReport.lon_int;
+
+#ifdef QT_QML_DEBUG
+    if(_simulate_motion == true) {
+        _createSimulatedMotion(follow_target);
+    }
+#endif
+
+    for (int i=0; i< vehicles.count(); i++) {
+        Vehicle* vehicle = qobject_cast<Vehicle*>(vehicles[i]);
+        if(vehicle->flightMode().compare(_followMeStr, Qt::CaseInsensitive) == 0) {
```

Comment:
Flight modes and flight mode names are firmware specific. The set of flight modes, their names and exactly what they do are opaque to QGC and differ between firmware stacks. This means you can't check for a specific flight mode like you are doing without breaking the mavlink generic rule. This needs to be moved to a Firmware plugin mechanism. I'm working on this right now. I'll include you in the pull so you can see what it ends up looking like.